web/src/main/java/org/cbioportal/web/util/AlterationCountUtil.java

```diff
@@ -0,0 +1,109 @@
+
+@Component
+public class AlterationCountUtil {
+    @Autowired
+    private AlterationCountService alterationCountService;
+    @Autowired
+    private GenePanelService genePanelService;
+    
+    public Pair<List<AlterationCountByGene>, Long> getAlterationCounts(
+        AlterationCountFilter alterationCountFilter,
+        AlterationCountType alterationCountType,
+        List<Integer> entrezGeneIds
+    ) {
+        // select all genes if no list of entrez gene ids provided
+        Select<Integer> geneSelect = (entrezGeneIds == null || entrezGeneIds.isEmpty()) ?
+            Select.all(): Select.byValues(entrezGeneIds);
+        List<MolecularProfileCaseIdentifier> molecularProfileCaseIdentifiers =
+            alterationCountFilter.getMolecularProfileCaseIdentifiers();
+        boolean includeFrequency = true;
+        boolean includeMissingAlterationsFromGenePanel = false;
+        AlterationFilter alterationFilter = alterationCountFilter.getAlterationFilter();
+
+         return alterationCountType.equals(AlterationCountType.SAMPLE) ?
+            alterationCountService.getSampleAlterationCounts(
+                molecularProfileCaseIdentifiers,
+                geneSelect,
+                includeFrequency,
+                includeMissingAlterationsFromGenePanel,
+                alterationFilter
+            ) :
+            alterationCountService.getPatientAlterationCounts(
+                molecularProfileCaseIdentifiers,
+                geneSelect,
+                includeFrequency,
+                includeMissingAlterationsFromGenePanel,
+                alterationFilter
+            );
+    }
+    
+    public Set<GenePanelToGene> getProfiledGenes(
+        AlterationCountFilter alterationCountFilter,
+        AlterationCountType alterationCountType,
+        List<Integer> entrezGeneIds
+    ) {
+        List<MolecularProfileCaseIdentifier> molecularProfileCaseIdentifiers =
+            alterationCountFilter.getMolecularProfileCaseIdentifiers();
+        Set<GenePanelToGene> profiledGenes;
+        List<GenePanelData> genePanelDataList = alterationCountType.equals(AlterationCountType.SAMPLE) ?
+            genePanelService.fetchGenePanelDataInMultipleMolecularProfiles(molecularProfileCaseIdentifiers) :
+            genePanelService.fetchGenePanelDataInMultipleMolecularProfilesByPatientIds(molecularProfileCaseIdentifiers);
+
+        if (genePanelDataList
+            .stream()
+            .anyMatch(
+                genePanelData -> genePanelData.getProfiled() && genePanelData.getGenePanelId() == null
+            )
+        ) {
+            // if there is at least one sample/patient with all genes profiles,
+            // then all genes are profiled for the current data set
+            profiledGenes = null;
+        } else {
+            List<String> profiledPanelIds =  genePanelDataList
+                .stream()
+                .filter(GenePanelData::getProfiled)
+                .map(GenePanelData::getGenePanelId)
```

Comment:
How come you're not doing the null check on genePanelId here?

Reply:
Good point. Seems like we should do null check.